### PR TITLE
[#65] Version number is now defined by the CMakeLists.txt file (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**
Verified that the package produces the correct information.

I need to run this against the test suite.

### CODE REVIEW

File: irods_specific_implementation/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,11 @@
-cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR) #CPACK_DEBIAN_<COMPONENT>_PACKAGE_NAME
+cmake_minimum_required(VERSION 3.11.0 FATAL_ERROR) #CPACK_DEBIAN_<COMPONENT>_PACKAGE_NAME
```
Comment (trel) [inline irods_specific_implementation/CMakeLists.txt:1]: do we need 3.11.0 actually minimum required?

File: irods_specific_implementation/constants.hpp [hunk 2]
```diff
@@ -5,7 +5,8 @@
 
 namespace irods::rest
 {
-    const std::string base_url = "/irods-rest/1.0.0";
+    // IRODS_CLIENT_VERSION is a macro that is defined by the CMakeLists.txt file.
+    const std::string base_url = IRODS_CLIENT_VERSION;
```
Comment (trel) [inline irods_specific_implementation/constants.hpp:9]: i think we still need the leading `/irods-rest/`?

**Review by trel — COMMENTED** [review]
i think the package requirements are still a little tight?   like irods-runtime and irods-server?

File: irods_specific_implementation/CMakeLists.txt (hunk 1 above)
Comment (korydraughn) [reply irods_specific_implementation/CMakeLists.txt:1]: i was getting a cmake error which lead this. this change isn't needed though. i'll revert it or use something in between.

File: irods_specific_implementation/constants.hpp (hunk 2 above)
Comment (korydraughn) [reply irods_specific_implementation/constants.hpp:9]: good catch

**Review by trel — COMMENTED** [review]
so many hard-coded versions, gah.    we'll think on whether we can reduce those somehow before 1.0.0...

File: irods_specific_implementation/CMakeLists.txt [hunk 3]
```diff
@@ -210,7 +218,7 @@ install(
   ${CMAKE_SOURCE_DIR}/packaging/irods_client_rest_cpp.conf.rsyslog
   ${CMAKE_SOURCE_DIR}/packaging/irods_client_rest_cpp.logrotate
   ${CMAKE_SOURCE_DIR}/packaging/irods_client_rest_cpp.json.template
-  ${CMAKE_SOURCE_DIR}/packaging/irods_client_rest_cpp_reverse_proxy.conf.template
+  ${CMAKE_BINARY_DIR}/packaging/irods_client_rest_cpp_reverse_proxy.conf.template
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply irods_specific_implementation/CMakeLists.txt:221]: That represents the build directory. The template is generated with the same version info defined by the CMakeLists.txt file.

[the remark this answers is not in the document]
Comment (korydraughn) [reply irods_specific_implementation/CMakeLists.txt:221]: The CMAKE_SOURCE_DIR contains the template.in file. The generated file will be under CMAKE_BINARY_DIR. Therefore, we have to reference that location for the packaging/installation.

**Comment by korydraughn** [comment]
i think this is ready to be merged.

is there anything else i need to do for this before adding the #'s?

**Comment by trel** [comment]
I think we're good - please # em.

**Comment by korydraughn** [comment]
added #

**Pull request merged**